tests/components/seventeentrack/test_sensor.py

```diff
@@ -2,283 +2,164 @@
+async def test_valid_config(
+    hass: HomeAssistant, mock_seventeentrack: AsyncMock
+) -> None:
     """Ensure everything starts correctly."""
-    assert await async_setup_component(hass, "sensor", VALID_CONFIG_MINIMAL)
-    await hass.async_block_till_done()
-    assert len(hass.states.async_entity_ids()) == len(ProfileMock.summary_data.keys())
+    await init_integration(hass, VALID_CONFIG_FULL)
+    assert len(hass.states.async_entity_ids()) == len(DEFAULT_SUMMARY.keys())
 
 
 async def test_invalid_config(hass: HomeAssistant) -> None:
     """Ensure nothing is created when config is wrong."""
-    assert await async_setup_component(hass, "sensor", INVALID_CONFIG)
+    await init_integration(hass, INVALID_CONFIG)
+    assert not hass.states.async_entity_ids("sensor")
+
 
+async def test_login_exception(
+    hass: HomeAssistant, mock_seventeentrack: AsyncMock
+) -> None:
+    """Ensure everything starts correctly."""
+    mock_seventeentrack.return_value.profile.login.side_effect = SeventeenTrackError(
+        "Error"
+    )
+    await init_integration(hass, VALID_CONFIG_FULL)
     assert not hass.states.async_entity_ids("sensor")
 
 
-async def test_add_package(hass: HomeAssistant) -> None:
+async def test_add_package(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory, mock_seventeentrack: AsyncMock
+) -> None:
     """Ensure package is added correctly when user add a new package."""
-    package = Package(
-        tracking_number="456",
-        destination_country=206,
-        friendly_name="friendly name 1",
-        info_text="info text 1",
-        location="location 1",
-        timestamp="2020-08-10 10:32",
-        origin_country=206,
-        package_type=2,
-    )
-    ProfileMock.package_list = [package]
+    package = get_package()
+    mock_seventeentrack.return_value.profile.packages.return_value = [package]
+    mock_seventeentrack.return_value.profile.summary.return_value = {}
 
-    await _setup_seventeentrack(hass)
+    await init_integration(hass, VALID_CONFIG_FULL)
     assert hass.states.get("sensor.seventeentrack_package_456") is not None
     assert len(hass.states.async_entity_ids()) == 1
 
-    package2 = Package(
+    package2 = get_package(
         tracking_number="789",
-        destination_country=206,
         friendly_name="friendly name 2",
         info_text="info text 2",
         location="location 2",
         timestamp="2020-08-10 14:25",
-        origin_country=206,
-        package_type=2,
     )
-    ProfileMock.package_list = [package, package2]
+    mock_seventeentrack.return_value.profile.packages.return_value = [package, package2]
 
-    await _goto_future(hass)
+    await _goto_future(hass, freezer)
 
     assert hass.states.get("sensor.seventeentrack_package_789") is not None
     assert len(hass.states.async_entity_ids()) == 2
 
 
-async def test_remove_package(hass: HomeAssistant) -> None:
+async def test_add_package_no_friendly_name(
```

Comment:
The name of the test doesn't reflect the code

Reply:
renamed